qemu/tests/format_disk.py

```diff
@@ -36,42 +36,45 @@ def run(test, params, env):
+            status, output = session.cmd_status_output(list_disk_cmd,
+                                                       timeout=cmd_timeout)
+            for i in re.findall("Disk*.(\d+)\s+Offline", output):
                 error.context("Set disk '%s' to online status" % i,
                               logging.info)
                 set_online_cmd = params.get("set_online_cmd") % i
-                s, o = session.cmd_status_output(set_online_cmd,
-                                                 timeout=cmd_timeout)
-                if s != 0:
-                    raise error.TestFail("Can not set disk online %s" % o)
+                status, output = session.cmd_status_output(set_online_cmd,
+                                                           timeout=cmd_timeout)
+                if status != 0:
+                    raise error.TestFail("Can not set disk online %s" % output)
 
         error.context("Create partition on disk", logging.info)
-        s, o = session.cmd_status_output(create_partition_cmd,
-                                         timeout=cmd_timeout)
-        if s != 0:
+        status, output = session.cmd_status_output(create_partition_cmd,
+                                                   timeout=cmd_timeout)
+        if status != 0:
             raise error.TestFail(
-                "Failed to create partition with error: %s" % o)
-        logging.info("Output of command of create partition on disk: %s" % o)
+                "Failed to create partition with error: %s" % output)
+        logging.info("Output of command of create partition on disk: %s" % output)
 
     format_cmd = params.get("format_cmd")
     if format_cmd:
         error.context("Format the disk with cmd '%s'" % format_cmd,
                       logging.info)
-        s, o = session.cmd_status_output(format_cmd,
-                                         timeout=cmd_timeout)
-        if s != 0:
-            raise error.TestFail("Failed to format with error: %s" % o)
-        logging.info("Output of format disk command: %s" % o)
+        status, output = session.cmd_status_output(format_cmd,
+                                                   timeout=cmd_timeout)
+        if status != 0:
+            raise error.TestFail("Failed to format with error: %s" % output)
+        logging.info("Output of format disk command: %s" % output)
 
     mount_cmd = params.get("mount_cmd")
+    show_dev_cmd = params.get("show_dev_cmd")
     if mount_cmd:
         error.context("Mount the disk with cmd '%s'" % mount_cmd, logging.info)
-        s, o = session.cmd_status_output(mount_cmd, timeout=cmd_timeout)
-        if s != 0:
-            raise error.TestFail("Failed to mount with error: %s" % o)
-        logging.info("Output of mount disk command: %s" % o)
+        status, output = session.cmd_status_output(mount_cmd, timeout=cmd_timeout)
+        if status != 0:
+            device_list = session.cmd_output_safe(show_dev_cmd)
```

Comment:
Hi @xutian 

Yes, params["show_dev_cmd"] can save 1 line here.

But I think the way that using params.get() is more easy to understand.

We can talk about it to decide which way is better.

Thanks.